### ISSUE TITLE
redisson3.16.1+springboot connection keep increase

### ISSUE THREAD

**Issue by smile326** [redisson/redisson#4379]
Hi, i use redis7.0, redisson 3.16.1 with springBoot, test with single server,  in the concurrent scenario, the number of connections keep to increase.

here is config,

```
<dependency>
    <groupId>org.redisson</groupId>
    <artifactId>redisson</artifactId>
    <version>3.16.1</version>
</dependency>
<dependency>
    <groupId>org.redisson</groupId>
    <artifactId>redisson-spring-data-24</artifactId>
    <version>3.16.1</version>
</dependency>

config.useSingleServer().setConnectionPoolSize(2);
config.useSingleServer().setConnectionMinimumIdleSize(2);
config.useSingleServer().setIdleConnectionTimeout(10000);
redisTemplate.setConnectionFactory(new RedissonConnectionFactory(Redisson.create(config)));
```

test code like this,

```
    @GetMapping("/redisTestRedissonAsync")
    public void redisTestRedissonAsync(){
        new Thread(new Runnable() {
            @Override
            public void run() {
                String key = "redission";
                RLock lock = redissonClient.getLock(key);
                try {
                    boolean result = lock.tryLock(15,600,TimeUnit.SECONDS);
                    LOGGER.error("result:{}",result);

                    key = String.valueOf(System.currentTimeMillis());
                    redisTemplate.opsForValue().set(key,key);
                    redisTemplate.expire(key,30, TimeUnit.SECONDS);
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        }).start();
    }
```

even set pool size 2, after application start success, 3 connection add in server,
then use ten threads and cycle ten times, total connection in server bigger than 3, and application print out:

```
[redisson-netty-2-10] o.r.connection.ClientConnectionsEntry    261 : new connection created: RedisConnection@880617262 [redisClient=[addr=redis://127.0.0.1:6379], channel=[id: 0xe1bbb6cc, L:/127.0.0.1:61758 - R:127.0.0.1/127.0.0.1:6379], currentCommand=CommandData [promise=RedissonPromise [promise=ImmediateEventExecutor$ImmediatePromise@3abadfd1(incomplete)], command=(PING), params=[], codec=org.redisson.client.codec.StringCodec]]
```

if cycle more times, connection keep increase, and when stop test, connection will not close automatically, even if wait more time than the value of idleConnectionTimeout.
but if test code like this:

```
new Thread(new Runnable() {
            @Override
            public void run() {
                String key = "redission";
                RLock lock = redissonClient.getLock(key);
                try {
                    boolean result = lock.tryLock(15,600,TimeUnit.SECONDS);
                    LOGGER.error("result:{}",result);
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        }).start();
```

or like this:

```
new Thread(new Runnable() {
            @Override
            public void run() {
                String key = "redission";
                RLock lock = redissonClient.getLock(key);
                try {
                    key = String.valueOf(System.currentTimeMillis());
                    redisTemplate.opsForValue().set(key,key);
                    redisTemplate.expire(key,30, TimeUnit.SECONDS);
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        }).start();
```

connection will keep 3, not increase.
It can stable reproduction.
I'm curious about the mechanism, and try to debug local, but have not find the reason.

**Comment by mrniko** [comment]
Can you reproduce it with the latest version?

**Comment by smile326** [comment]
> Can you reproduce it with the latest version?

use redission 3.17.1, test again, the number of connections remains stable.

So version 3.16 is the possibility of connection leaks? And 3.17 has repaired something, if that maybe we should update version to fix this.

**Comment by mrniko** [comment]
This leak was fixed in 3.16.3 version.

**Comment by smile326** [comment]
> This leak was fixed in 3.16.3 version.

get it, thanks very much.
